UKHO.PeriodicOutputService/UKHO.BESS.ConfigurationService/Services/ConfigurationService.cs

```diff
@@ -16,80 +18,134 @@ public class ConfigurationService : IConfigurationService
         {
             this.azureBlobStorageClient = azureBlobStorageClient ?? throw new ArgumentNullException(nameof(azureBlobStorageClient));
             this.azureTableStorageHelper = azureTableStorageHelper ?? throw new ArgumentNullException(nameof(azureTableStorageHelper));
             this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this.configValidator = configValidator ?? throw new ArgumentNullException(nameof(configValidator));
         }
 
         public void ProcessConfigs()
         {
             try
             {
-                IList<BessConfig> bessConfigs = new List<BessConfig>();
+                var configsInContainer = azureBlobStorageClient.GetConfigsInContainer();
+                if (!configsInContainer.Any())
+                {
+                    logger.LogWarning(EventIds.BessConfigsNotFound.ToEventId(), "Bess configs not found | _X-Correlation-ID : {CorrelationId}", CommonHelper.CorrelationID);
+                }
+
+                logger.LogInformation(EventIds.BessConfigsProcessingStarted.ToEventId(), "Bess configs processing started, Total configs file count : {count}  | _X-Correlation-ID : {CorrelationId}", configsInContainer.Keys.Count, CommonHelper.CorrelationID);
 
-                Dictionary<string, string> configs = azureBlobStorageClient.GetConfigsInContainer();
+                IList<BessConfig> bessConfigs = new List<BessConfig>();
 
-                logger.LogInformation(EventIds.BessConfigsProcessingStarted.ToEventId(), "Bess configs processing started, Total configs count:{count}  | _X-Correlation-ID : {CorrelationId}", configs.Keys.Count, CommonHelper.CorrelationID);
+                int configsWithInvalidAttributeCount = 0;
+                int deserializedConfigsCount = 0;
 
-                if (configs.Any())
+                foreach (string fileName in configsInContainer.Keys.ToList())
                 {
-                    foreach (string fileName in configs.Keys.ToList())
+                    string content = configsInContainer[fileName];
+
+                    var deserializedConfigs = DeserializeConfig(content, fileName);
+
+                    foreach (BessConfig deserializedConfig in deserializedConfigs)
                     {
-                        string content = configs[fileName];
+                        deserializedConfigsCount++;
+                        deserializedConfig.FileName = fileName; //for logging
 
-                        bool isValidJson = IsValidJson(content, fileName);
+                        ValidationResult results = configValidator.Validate(deserializedConfig);
 
-                        if (isValidJson)
+                        if (!results.IsValid)
                         {
-                            IList<BessConfig> bessConfig = JsonConvert.DeserializeObject<List<BessConfig>>(content)!;
+                            configsWithInvalidAttributeCount++;
+                            string errors = string.Empty;
```

Comment:
Can we use StringBuilder here ?
```suggestion
                            var errors = new StringBuilder();
```

Reply:
Resolved. Thanks